lib/js/dom.nim

```diff
@@ -400,6 +400,17 @@ type
     once*: bool
     passive*: bool
 
+  MathLib* = ref object
```

Comment:
The functions and types you added here aren't part of the DOM. They should be moved to a different module.

Reply:
Not if you treat the DOM as the larger API surface of the browser. What would be a suitable name for this other module?

Reply:
In Javascript there is the Core (that is, the part of the stdlib that is requested in all environments, even outside browers), the DOM and the BOM (Browser object model). https://vkanakaraj.wordpress.com/2009/12/18/javascript-vs-dom-vs-bom-relationship-explained/

Objects such as Math, Date and so on live in the code JS library

Reply:
``jscore`` then?

Reply:
Could make sense, although I expect most of this functionality to be available in the usual parts of the stdlib with the usual Nim names (e.g. under the modules `math` or `times` or `random`)

Reply:
The point of these exposed APIs is that they allow you to call the "low-level" native methods of the JavaScript environment from jsffi-based code, which adds only a very thin layer on top of JavaScript.

Nim modules such as `math` and `times` have a slightly different goal. For all intents and purposes, they try to preserve the exact behavior of the C run-time of Nim, which often means having a thicker layer responsible for converting between the Nim and JavaScript data types and formats.

Reply:
Agreed. I would use `jscore`, `dom` and `bom` (the latter for things such as navigator, history etc. if we have it)